### PR TITLE
refactor(sample add filters and tests)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adding filters and filter tests for the sample model

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/filters/test_status_samples_filters.py [hunk 1]
```diff
@@ -418,12 +454,18 @@ def test_filter_samples_is_not_prepared(
...
+    assert len(samples.all()) == 1
+
+    # THEN the sample should have not have a prepared at date
+    assert samples.all()[0].prepared_at is None
 
 
 def test_filter_get_samples_by_internal_id(
     store_with_a_sample_that_has_many_attributes_and_one_without: Store,
-    sample_id: str = StoreConftestFixture.INTERNAL_ID_SAMPLE_WITH_ATTRIBUTES.value,
+    internal_id: str = StoreConftestFixture.INTERNAL_ID_SAMPLE_WITH_ATTRIBUTES.value,
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_samples_filters.py:468]:
```suggestion
    sample_internal_id: str = StoreConftestFixture.INTERNAL_ID_SAMPLE_WITH_ATTRIBUTES.value,
```

File: tests/store/filters/test_status_samples_filters.py [hunk 2]
```diff
@@ -434,14 +476,20 @@ def test_filter_get_samples_by_internal_id(
         samples=store_with_a_sample_that_has_many_attributes_and_one_without._get_query(
             table=Sample
         ),
-        internal_id=sample_id,
+        internal_id=internal_id,
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_samples_filters.py:479]:
```suggestion
        internal_id=sample_internal_id,
```

File: tests/store/filters/test_status_samples_filters.py [hunk 3]
```diff
@@ -464,7 +512,13 @@ def test_filter_get_samples_by_entry_id(
     assert isinstance(samples, Query)
 
     # THEN samples should contain the test sample
-    assert samples.all() and len(samples.all()) == 1
+    assert samples.all()
+
+    # THEN samples should contain one sample
+    assert len(samples.all()) == 1
+
+    # THEN the sample should have the correct id
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_samples_filters.py:520]:
```suggestion
    # THEN the sample should have the correct entry id
```

File: tests/store/filters/test_status_samples_filters.py [hunk 4]
```diff
@@ -532,4 +598,66 @@ def test_filter_get_samples_by_subject_id(
...
+        customer_ids=[customer_id],
+    )
+
+    # ASSERT that samples is a query
+    assert isinstance(samples, Query)
+
+    # THEN samples should contain the test sample
+    assert samples.all()
+
+    # THEN samples should contain one sample
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_samples_filters.py:631]:
```suggestion
    # THEN samples should contain two samples
```

File: tests/store/filters/test_status_samples_filters.py [hunk 5]
```diff
@@ -532,4 +598,66 @@ def test_filter_get_samples_by_subject_id(
...
+    assert len(samples.all()) == 1
+
+    # THEN the sample should have the correct name enquiry id
```
Comment (henrikstranneheim) [inline tests/store/filters/test_status_samples_filters.py:662]:
```suggestion
    # THEN the sample should have the correct name
```

File: cg/store/filters/status_sample_filters.py [hunk 6]
```diff
@@ -87,9 +87,9 @@ def filter_samples_do_not_invoice(samples: Query, **kwargs) -> Query:
     return samples.filter(Sample.no_invoice.is_(True))
 
 
-def filter_samples_by_customer_id(samples: Query, customer_id: str, **kwargs) -> Query:
+def filter_samples_by_customer_id(samples: Query, customer_ids: List[int], **kwargs) -> Query:
```
Comment (henrikstranneheim) [inline cg/store/filters/status_sample_filters.py:90]: Do you not also have to update the call to this function and the apply_function to not brake the code?

File: cg/store/filters/status_sample_filters.py [hunk 7]
```diff
@@ -132,6 +132,17 @@ def filter_samples_is_not_tumour(samples: Query, **kwargs) -> Query:
     return samples.filter(Sample.is_tumour.is_(False))
 
 
+def filter_samples_by_name_enquiry(samples: Query, name_enquiry: str, **kwargs) -> Query:
```
Comment (seallard) [inline cg/store/filters/status_sample_filters.py:135]: I think "enquiry" is confusing and would like the name to be easier to understand. Something like:
```python
def filter_samples_by_name_pattern(samples: Query, name_pattern: str, **kwargs) -> Query:
    """Return samples matching the name pattern."""
...
```

**Review by seallard — APPROVED** [review]

**Comment by ChrOertlin** [comment]
Stage:

Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 27.1.18


Production:

Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/hiseq.clinical
done.
Log deploy... done.
cg, version 27.1.18

**Pull request merged**